libindy/src/commands/mod.rs

```diff
@@ -202,31 +212,54 @@ impl CommandExecutor {
                         }
                         Command::Payments(cmd) => {
                             debug!("PaymentsCommand command received");
-                            payments_command_executor.execute(cmd);
+                            payments_command_executor.execute(cmd).await;
                         }
                         Command::Cache(cmd) => {
                             debug!("CacheCommand command received");
-                            cache_command_executor.execute(cmd);
+                            cache_command_executor.execute(cmd).await;
                         }
                         Command::Metrics(cmd) => {
                             debug!("MetricsCommand command received");
                             metrics_command_executor.execute(cmd);
                         }
                         Command::Exit => {
                             debug!("Exit command received");
-                            break
                         }
                     }
+
                     metrics_service.cmd_executed(cmd_index,
                                                  get_cur_time() - start_execution_ts);
```

Comment:
Now executed time includes waiting time as well. This is not a blocker for this PR, but as a next step I would suggest to spend at least some time to look into possibility of tracking executing and waiting time separately, including whether we need this separation.

Reply:
FYI @Toktar